### PR TITLE
Add support for a worker variation of the Edxapp.

### PR DESCRIPTION
**Pull request by feanil**

### CODE REVIEW

File: playbooks/roles/edxapp/tasks/cms.yml [hunk 1]
```diff
@@ -18,8 +18,10 @@
   - update
 
 - include: ../../nginx/tasks/nginx_site.yml state=link site_name=cms
+  when: celery_worker is not defined
```
Comment (jbau) [inline playbooks/roles/edxapp/tasks/cms.yml:21]: I think this play is already conditionalized on whether the nginx role has run or not:

```
- name: Copying nginx config {{ site_name }}
  template: src={{ item }} dest=/etc/nginx/sites-available/{{ site_name }}
  first_available_file:
  - "{{ local_dir }}/nginx/templates/{{ site_name }}.j2"
  # seems like paths in first_available_file must be relative to the playbooks dir
  - "roles/nginx/templates/{{ site_name }}.j2" 
  notify: restart nginx
  when_set: $nginx_role_run
```

File: playbooks/roles/edxapp/tasks/cms.yml [hunk 2]
```diff
@@ -30,4 +32,8 @@
   - update
 
 # Creates LMS upstart file
```
Comment (jbau) [inline playbooks/roles/edxapp/tasks/cms.yml]: can we fix the comment (i know, nitpicky, but we might as well pick them off one by one)

File: playbooks/roles/edxapp/tasks/cms.yml (hunk 1 above)
Comment (jbau) [inline playbooks/roles/edxapp/tasks/cms.yml:21]: but that's more FYI.  I can see how setting celery_worker is useful in this task file.

File: playbooks/roles/edxapp/tasks/cms.yml [hunk 3]
```diff
@@ -30,4 +32,8 @@
...
-- include: upstart.yml service_variant=cms
+- include: upstart.yml basename=cms
+  when: celery_worker is not defined
+
+- include: upstart.yml basename=edx-worker-cms
```
Comment (jbau) [inline playbooks/roles/edxapp/tasks/cms.yml:38]: just curious.  Does the cms actually have celery jobs?

File: playbooks/roles/edxapp/templates/edx-workers.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+   start edx-worker-cms QUEUE=low CONCURRENCY=1 SERVICE_VARIANT=cms
+   start edx-worker-cms QUEUE=default CONCURRENCY=3 SERVICE_VARIANT=cms
+   start edx-worker-cms QUEUE=high CONCURRENCY=4 SERVICE_VARIANT=cms
+
+   start edx-worker-lms QUEUE=low CONCURRENCY=1 SERVICE_VARIANT=lms
+   start edx-worker-lms QUEUE=default CONCURRENCY=3 SERVICE_VARIANT=lms
+   start edx-worker-lms QUEUE=high CONCURRENCY=4 SERVICE_VARIANT=lms
+
+   start edx-worker-lms-xml QUEUE=low CONCURRENCY=1 SERVICE_VARIANT=lms-xml
```
Comment (jbau) [inline playbooks/roles/edxapp/templates/edx-workers.conf.j2:24]: can we parameterize this on service_variants_enabled?  (Or I can)

**Comment by jbau** [comment]
running this now...
it also seems like we should be putting the ruby and node installs and the theme, etc, under the `celery_worker` in deploy.yml.

I wonder if it's better just to write a new worker-deploy.yml given the number of changes.

**Comment by jbau** [comment]
I added parameterization of edx-workers.conf on service_variants_enabled, and ran this branch against our worker machines, which seems to work fine.

**Comment by feanil** [comment]
Any objections to rebasing this and merging to master?

**Comment by jbau** [comment]
Not here.

**Comment by feanil** [comment]
Rebased and merged via: https://github.com/edx/configuration/pull/164

**Pull request closed without merging**